applications/FSIapplication/python_scripts/trilinos_partitioned_fsi_dirichlet_neumann_solver.py

```diff
@@ -212,79 +115,112 @@ def _InitializeDirichletNeumannInterface(self):
+            KratosMultiphysics.VAUX_EQ_TRACTION,
+            KratosStructural.POINT_LOAD)
 
         # Solve the structure problem
-        self.structure_solver.SolveSolutionStep()
-
+        is_converged = self.structure_solver.SolveSolutionStep()
+        if not is_converged:
+            self._PrintWarningOnRankZero("Structure solver did not converge.")
 
-    # TODO: UPDATE THIS ONCE THE DOUBLE SIDED SURFACES MAPPING IS AVAILABLE IN MAPPING APPLICATION
     def _SolveStructureDoubleFaced(self):
-
-        # Transfer fluid reaction from both sides to the structure interface
-        keep_sign = False
-        distribute_load = True
-        self.interface_mapper.PositiveFluidToStructure_VectorMap(KratosMultiphysics.REACTION,
-                                                                 KratosFSI.POSITIVE_MAPPED_VECTOR_VARIABLE,
-                                                                 keep_sign,
-                                                                 distribute_load)
-        self.interface_mapper.NegativeFluidToStructure_VectorMap(KratosMultiphysics.REACTION,
-                                                                 KratosFSI.NEGATIVE_MAPPED_VECTOR_VARIABLE,
-                                                                 keep_sign,
-                                                                 distribute_load)
-
-        # Add the two faces contributions to the POINT_LOAD variable
-        # TODO: Add this to the variables utils
-        for node in self._GetStructureInterfaceSubmodelPart().Nodes:
-            pos_face_force = node.GetSolutionStepValue(KratosFSI.POSITIVE_MAPPED_VECTOR_VARIABLE)
-            neg_face_force = node.GetSolutionStepValue(KratosFSI.NEGATIVE_MAPPED_VECTOR_VARIABLE)
-            node.SetSolutionStepValue(KratosStructural.POINT_LOAD, 0, pos_face_force+neg_face_force)
+        # Set the redistribution settings
+        redistribution_tolerance = 1e-8
+        redistribution_max_iters = 50
+
+        # Convert the nodal reaction to traction loads before transfering
+        KratosFSI.VariableRedistributionUtility.DistributePointValues(
+            self._GetFluidPositiveInterfaceSubmodelPart(),
+            KratosMultiphysics.REACTION,
+            KratosMultiphysics.VAUX_EQ_TRACTION,
+            redistribution_tolerance,
+            redistribution_max_iters)
+
+        KratosFSI.VariableRedistributionUtility.DistributePointValues(
+            self._GetFluidNegativeInterfaceSubmodelPart(),
+            KratosMultiphysics.REACTION,
+            KratosMultiphysics.VAUX_EQ_TRACTION,
+            redistribution_tolerance,
+            redistribution_max_iters)
+
+        # Transfer fluid tractions to the structure interface
+        self.pos_interface_mapper.Map(KratosMultiphysics.VAUX_EQ_TRACTION,
+                                      KratosMultiphysics.VAUX_EQ_TRACTION,
+                                      KratosMapping.Mapper.SWAP_SIGN,
```

Comment:
this is not working for two reasons:
the correct python syntax is:
`KratosMapping.Mapper.SWAP_SIGN | KratosMapping.Mapper.ADD_VALUES)`

Furthermore you should not specify `ADD_VALUES` for the first mapper, only for the following ones
The first one should actually overwrite the previous values

Reply:
you can take a look at the [wiki](https://github.com/KratosMultiphysics/Kratos/wiki/How-to-do-Mapping-between-nonmatching-meshes#using-the-mapping-functions)

Reply:
Done. Thanks for noticing!